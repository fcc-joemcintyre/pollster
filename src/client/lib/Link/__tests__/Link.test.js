/* eslint-env jest */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { PlainLink } from '..';

const theme = {
};

test ('PlainLink default render', () => {
  const component = renderer.create (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <PlainLink to='#'>text</PlainLink>
      </ThemeProvider>
    </BrowserRouter>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainLink renders with color', () => {
  const component = renderer.create (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <PlainLink to='$' c='#000000'>text</PlainLink>
      </ThemeProvider>
    </BrowserRouter>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});
