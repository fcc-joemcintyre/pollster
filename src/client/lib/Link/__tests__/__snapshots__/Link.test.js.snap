// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainLink default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0"
  href="/test"
  onClick={[Function]}
>
  text
</a>
`;

exports[`PlainLink renders with color 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000000;
}

.c0:visited: {
  color: #000000;
}

<a
  c="#000000"
  className="c0"
  href="/test"
  onClick={[Function]}
>
  text
</a>
`;
